components/shared/mentors.tsx

```diff
@@ -1,103 +1,144 @@
+          },
+        }
+      );
+
+      const transformedData = Array?.isArray(response.data)
+        ? response.data
+        : response.data.items || [];
+
+      setMentors(
+        transformedData.map((mentor: any) => ({
+          id: mentor.id,
+          name: mentor.user.name,
+          expertise: mentor.expertise,
+          age: mentor.age,
+          gender: mentor.gender,
+          location: mentor.location,
+          availability: mentor.availability.startDate,
+          isActive: mentor.isActive,
+          profileImage: mentor.user.imageUrl || "",
+        }))
+      );
+    } catch (error) {
+      console.error("Error fetching mentors:", error);
+    }
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    fetchMentors();
+  }, []);
+
+  const handleDelete = async (id: string | number) => {
+    console.log("Deleting mentor with id:", id);
+    if (!id) {
+      console.error("Delete failed: Invalid mentor ID");
+      return;
+    }
+
+    try {
+      await axios.delete(
+        `${process.env.NEXT_PUBLIC_API_URL}/admin/mentors/${id}`,
+        {
+          headers: {
+            Authorization: `Bearer ${`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIzZjE0MjE5ZC0zNGNhLTRiZDQtYWQyNy1hNjE4Y2E0YWEyZDAiLCJlbWFpbCI6ImplcmloYWdiakBnbWFpbC5jb20iLCJpbWFnZVVybCI6bnVsbCwiYWNjb3VudHMiOlt7ImlkIjoiODc4ODhmODAtZGI0Yi00OWUyLTgxMDQtMTMyYjMzOTU4NmQxIiwibmFtZSI6IkplcnVzYWxlbSJ9XSwicm9sZXMiOlsiT1dORVIiXSwiaWF0IjoxNzM0MjAyODU4fQ.7LYXWzv0phj_Xdg3YT-dkKGAovbhNlA16zsP3qUghvU`}`, // Include the Bearer token here
+          },
+        }
+      );
+      setMentors((prev) => prev.filter((mentor) => mentor.id !== id));
+      console.log("Mentor deleted successfully");
+    } catch (error) {
+      console.error("Error deleting mentor:", error);
+    }
+  };
   return (
     <div className="flex-1 p-4 bg-secondary dark:bg-gray-900">
-      <div className="space-y-6  bg-white p-6 rounded-lg">
-        <div className="flex items-center justify-between ">
+      <div className="space-y-6 bg-white p-6 rounded-lg">
+        <div className="flex items-center justify-between">
           <h1 className="text-2xl font-semibold">Mentors</h1>
           <InviteMentorDialog />
         </div>
         <DataTable
-          data={ALL_MENTORS}
+          apiUrl={`${process.env.NEXT_PUBLIC_API_URL}/admin/mentors/`}
           columns={columns}
-          searchFields={[
-            "name",
-            "email",
-            "specialization",
-            "location",
-            "age",
-            "gender",
-            "status",
-          ]}
+          searchFields={["name", "age", "gender", "location", "isActive"]}
```

Comment:
🤔put it some where , and add here the var name kindly. ["name", "age", "gender", "location", "isActive"] you can reuse it.